app/src/main/java/com/mapbox/maps/testapp/examples/viewport/ViewportShowcaseActivity.kt

```diff
@@ -0,0 +1,237 @@
+  private lateinit var viewportButton: Button
+  private val pixelDensity = Resources.getSystem().displayMetrics.density
+
+  private val paddedFollowingEdgeInsets = EdgeInsets(
+    0.0,
+    0.0,
+    120.0 * pixelDensity,
+    0.0
+  )
+
+  private val notPaddedEdgeInsets: EdgeInsets by lazy {
+    EdgeInsets(
+      0.0,
+      0.0,
+      0.0,
+      0.0
+    )
+  }
+  private var followingEdgeInsets = paddedFollowingEdgeInsets
+    set(value) {
+      field = value
+      viewportDataSource.followingPadding = value
+      viewportDataSource.evaluate()
+    }
+  private val overviewEdgeInsets: EdgeInsets by lazy {
+    EdgeInsets(
+      40.0 * pixelDensity,
+      40.0 * pixelDensity,
+      40.0 * pixelDensity,
+      40.0 * pixelDensity
+    )
+  }
+  private val overviewPadding: EdgeInsets by lazy {
+    EdgeInsets(
+      140.0 * pixelDensity,
+      40.0 * pixelDensity,
+      120.0 * pixelDensity,
+      40.0 * pixelDensity
+    )
+  }
+  private val landscapeOverviewPadding: EdgeInsets by lazy {
+    EdgeInsets(
+      30.0 * pixelDensity,
+      380.0 * pixelDensity,
+      20.0 * pixelDensity,
+      20.0 * pixelDensity
+    )
+  }
+  private val followingPadding: EdgeInsets by lazy {
+    EdgeInsets(
+      180.0 * pixelDensity,
+      40.0 * pixelDensity,
+      150.0 * pixelDensity,
+      40.0 * pixelDensity
+    )
+  }
+  private val landscapeFollowingPadding: EdgeInsets by lazy {
+    EdgeInsets(
+      30.0 * pixelDensity,
+      380.0 * pixelDensity,
+      110.0 * pixelDensity,
+      40.0 * pixelDensity
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val binding = ActivityViewportAnimationBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+    viewportButton = binding.switchButton
+    mapView = binding.mapView
+    mapboxMap = binding.mapView.getMapboxMap().apply {
+      addOnStyleLoadedListener {
+        setCamera(
+          CameraOptions.Builder()
+            .zoom(14.0)
+            .center(Point.fromLngLat(POINT_LNG, POINT_LAT))
+            .build()
+        )
+        binding.mapView.location.apply {
+          val provider = FakeLocationProvider()
+          latestLocation = Location(provider.javaClass.simpleName)
+          setLocationProvider(provider)
+          updateSettings {
+            locationPuck = LocationPuck2D(
+              bearingImage = AppCompatResources.getDrawable(
+                this@ViewportShowcaseActivity,
+                R.drawable.mapbox_mylocation_icon_bearing,
+              )
+            )
+          }
+          addOnIndicatorPositionChangedListener {
+            latestLocation.latitude = it.latitude()
+            latestLocation.longitude = it.longitude()
+          }
+          addOnIndicatorBearingChangedListener {
+            latestLocation.bearing = it.toFloat()
+            viewportDataSource.onLocationChanged(latestLocation)
+            viewportCamera.resetFrame()
+          }
```

Comment:
We have two callbacks from the SDK when location changes but only in bearing we are updating the viewportdatasource. This is probably due to the 2 independent callbacks. Would it make sense to have an unified callback that both provides location as bearing in one go?

Reply:
There are cases where these two updates might come independently, for example, if we want to update the bearing according to the magnetic sensor on the phone instead of the one from GPS location.

However I think we should update viewPortDataSource in both listeners.